bin/proving-service/src/proxy/status.rs

```diff
@@ -1,103 +1,200 @@
+        },
+    },
+    proxy::LoadBalancerState,
 };
-use serde::Serialize;
-use tonic::async_trait;
-use tracing::error;
-
-use super::worker::WorkerHealthStatus;
-use crate::{commands::worker::ProverType, proxy::LoadBalancerState};
-
-// Status of a worker
-#[derive(Debug, Serialize)]
-pub struct WorkerStatus {
-    address: String,
-    version: String,
-    status: WorkerHealthStatus,
-}
 
-/// Status of the proxy
-#[derive(Debug, Serialize)]
-pub struct ProxyStatus {
-    version: String,
-    prover_type: ProverType,
-    workers: Vec<WorkerStatus>,
-}
+/// Update cache every 5 seconds
+const CACHE_UPDATE_INTERVAL_SECS: u64 = 5;
+
+/// Cached proxy status response
+type CachedStatus = ProxyStatusResponse;
+
+// PROXY STATUS SERVICE
+// ================================================================================================
 
-/// Service that handles status requests
+/// gRPC service that implements Pingora's Service trait
 pub struct ProxyStatusService {
     load_balancer: Arc<LoadBalancerState>,
+    port: u16,
+    status_tx: watch::Sender<CachedStatus>,
+    update_interval: Duration,
+}
+
+/// Internal gRPC service implementation
+pub struct ProxyStatusGrpcService {
+    status_rx: watch::Receiver<CachedStatus>,
 }
 
 impl ProxyStatusService {
-    pub fn new(load_balancer: Arc<LoadBalancerState>) -> Self {
-        Self { load_balancer }
+    pub async fn new(load_balancer: Arc<LoadBalancerState>, port: u16) -> Self {
+        // Create initial status
+        let initial_status = generate_status(&load_balancer).await;
+
+        let (status_tx, _) = watch::channel(initial_status);
+        let update_interval = Duration::from_secs(CACHE_UPDATE_INTERVAL_SECS);
+
+        Self {
+            load_balancer,
+            port,
+            status_tx,
+            update_interval,
+        }
     }
+}
 
-    async fn handle_request(&self, session: &mut ServerSession) -> Result<()> {
-        let workers = self.load_balancer.workers.read().await;
-        let worker_statuses: Vec<WorkerStatus> = workers
-            .iter()
-            .map(|w| WorkerStatus {
-                address: w.address(),
-                version: w.version().to_string(),
-                status: w.health_status().clone(),
-            })
-            .collect();
-
-        let status = ProxyStatus {
-            version: env!("CARGO_PKG_VERSION").to_string(),
-            prover_type: self.load_balancer.supported_prover_type,
-            workers: worker_statuses,
-        };
+/// Generates a new status and stores it in the cache.
+async fn generate_and_store_status(
+    load_balancer: &LoadBalancerState,
+    status_tx: &watch::Sender<CachedStatus>,
+) {
+    let status = generate_status(load_balancer).await;
+
+    // This will notify all receivers of the new status
+    let _ = status_tx.send(status);
+}
+
+/// Generates a new status from the load balancer.
+async fn generate_status(load_balancer: &LoadBalancerState) -> ProxyStatusResponse {
```

Comment:
A few nits:
- I would name this `build_status()`.
- This looks like a helper function - let's move it into the `HELPER FUNCTIONS` section at the end of the file.
- Could this be a method on `ProxyStatusService`?

Also, let's improve doc comments here to explain what's going on. My understanding is that this doesn't make any requests to the work instances, we just read the currently cached statuses of each worker - correct?